app/controllers/courses_controller.rb

```diff
@@ -278,6 +280,27 @@ def update_courses_wikis
     @course.update_wikis(new_wikis)
   end
 
+  def course_wiki_namespaces_params
+    params.require(:course).permit(namespaces: [])[:namespaces]
+  end
+
+  def update_course_wiki_namespaces
+    namespaces = course_wiki_namespaces_params
+
+    @course.courses_wikis.each do |course_wiki|
+      wiki_domain = course_wiki.wiki.domain
+      cw_ns = []
+      namespaces.each do |ns|
+        ns_wiki_domain = ns.split('-', 2)[0]
```

Comment:
This code would be easier to understand with some comments showing what shape `namespaces` takes.